cluster/k8s-1.9.3/provider.sh

```diff
@@ -2,73 +2,30 @@
-    OPTIONS="-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i ${KUBEVIRT_PATH}cluster/k8s-1.9.3/vagrant.key -P 2201"
+    chmod 0600 ${KUBEVIRT_PATH}cluster/vagrant.key
+    OPTIONS="-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i ${KUBEVIRT_PATH}cluster/vagrant.key -P 2201"
 
     # Copy k8s config and kubectl
-    scp ${OPTIONS} vagrant@127.0.0.1:/usr/bin/kubectl ${KUBEVIRT_PATH}cluster/k8s-1.9.3/.kubectl
+    scp ${OPTIONS} vagrant@127.0.0.1:/usr/bin/kubectl ${KUBEVIRT_PATH}cluster/$PROVIDER/.kubectl
     chmod u+x cluster/vagrant-kubernetes/.kubectl
-    scp ${OPTIONS} vagrant@127.0.0.1:/etc/kubernetes/admin.conf ${KUBEVIRT_PATH}cluster/k8s-1.9.3/.kubeconfig
+    scp ${OPTIONS} vagrant@127.0.0.1:/etc/kubernetes/admin.conf ${KUBEVIRT_PATH}cluster/$PROVIDER/.kubeconfig
 
     # Set server and disable tls check
-    export KUBECONFIG=${KUBEVIRT_PATH}cluster/k8s-1.9.3/.kubeconfig
-    ${KUBEVIRT_PATH}cluster/k8s-1.9.3/.kubectl config set-cluster kubernetes --server=https://127.0.0.1:8443
-    ${KUBEVIRT_PATH}cluster/k8s-1.9.3/.kubectl config set-cluster kubernetes --insecure-skip-tls-verify=true
+    export KUBECONFIG=${KUBEVIRT_PATH}cluster/$PROVIDER/.kubeconfig
+    ${KUBEVIRT_PATH}cluster/$PROVIDER/.kubectl config set-cluster kubernetes --server=https://$(_main_ip):8443
+    ${KUBEVIRT_PATH}cluster/$PROVIDER/.kubectl config set-cluster kubernetes --insecure-skip-tls-verify=true
 
     # Make sure that local config is correct
     prepare_config
 }
-
-function prepare_config() {
-    BASE_PATH=${KUBEVIRT_PATH:-$PWD}
-    cat >hack/config-provider-k8s-1.9.3.sh <<EOF
-master_ip=$(_main_ip)
-docker_tag=devel
-kubeconfig=${BASE_PATH}/cluster/k8s-1.9.3/.kubeconfig
-docker_prefix=localhost:5000/kubevirt
-manifest_docker_prefix=registry:5000/kubevirt
-EOF
-}
-
-function build() {
-    # Build everyting and publish it
-    ${KUBEVIRT_PATH}hack/dockerized "DOCKER_TAG=${DOCKER_TAG} PROVIDER=${PROVIDER} ./hack/build-manifests.sh"
-    make build docker publish
-
-    # Make sure that all nodes use the newest images
-    container=""
-    container_alias=""
-    for arg in ${docker_images}; do
-        local name=$(basename $arg)
-        container="${container} ${manifest_docker_prefix}/${name}:${docker_tag}"
-        container_alias="${container_alias} ${manifest_docker_prefix}/${name}:${docker_tag} kubevirt/${name}:${docker_tag}"
-    done
-    local num_nodes=${VAGRANT_NUM_NODES-0}
-    num_nodes=$((num_nodes + 1))
-    for i in $(seq 1 ${num_nodes}); do
-        ${_cli} ssh "node$(printf "%02d" ${i})" "echo \"${container}\" | xargs --max-args=1 sudo docker pull"
-        ${_cli} ssh "node$(printf "%02d" ${i})" "echo \"${container_alias}\" | xargs --max-args=2 sudo docker tag"
-    done
-}
-
-function _kubectl() {
-    export KUBECONFIG=${KUBEVIRT_PATH}cluster/k8s-1.9.3/.kubeconfig
-    ${KUBEVIRT_PATH}cluster/k8s-1.9.3/.kubectl "$@"
-}
-
-function down() {
-    ${_cli} rm
```

Comment:
The right prefix needs to be used to delete the right cluster.

Reply:
I removed this one, new file has this one
```
function down() {
    ${_cli} rm --prefix $PROVIDER
}
```